Assets/BossRoom/Scripts/Editor/GuidScriptableObjectEditor.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using UnityEditor;
+using UnityEngine;
+
+namespace BossRoom.Scripts.Editor
+{
+    /// <summary>
+    /// Custom inspector for <see cref="GuidScriptableObject"/>, and is responsible for assigning GUID.
+    /// </summary>
+    [CustomEditor(typeof(GuidScriptableObject), true)]
+    public class GuidScriptableObjectEditor : UnityEditor.Editor
+    {
+        public override void OnInspectorGUI()
+        {
+            base.OnInspectorGUI();
+
+            var guidScriptableObject = (GuidScriptableObject)target;
+
+            var guidScriptableObjectSerializedObject = new SerializedObject(guidScriptableObject);
+
+            var serializedGuidProperty = guidScriptableObjectSerializedObject.FindProperty("m_Guid");
```

Comment:
see if we can simplify this with OnValidate

Reply:
Yes, this script has since been removed and byte arrays are now populated on `OnValidate` of `GuidScriptableObject.`